azurerm/resource_arm_healthcare_service.go

```diff
@@ -0,0 +1,436 @@
+	}
+
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for the deleting Healthcare Service %q (Resource Group %q): %+v", name, resGroup, err)
+	}
+
+	return nil
+}
+
+func expandAzureRMhealthcareapisAccessPolicyEntries(d *schema.ResourceData) *[]healthcareapis.ServiceAccessPolicyEntry {
+	accessPolicyObjectIds := d.Get("access_policy_object_ids").([]interface{})
+	svcAccessPolicyArray := make([]healthcareapis.ServiceAccessPolicyEntry, 0)
+
+	for _, objectId := range accessPolicyObjectIds {
+		objectIdsStr := objectId.(string)
+		svcAccessPolicyObjectId := healthcareapis.ServiceAccessPolicyEntry{ObjectID: &objectIdsStr}
+		svcAccessPolicyArray = append(svcAccessPolicyArray, svcAccessPolicyObjectId)
+	}
+
+	return &svcAccessPolicyArray
+}
+
+func expandAzureRMhealthcareapisCorsConfiguration(d *schema.ResourceData) *healthcareapis.ServiceCorsConfigurationInfo {
+	corsConfigRaw := d.Get("cors_configuration").([]interface{})
+
+	if len(corsConfigRaw) == 0 {
+		return &healthcareapis.ServiceCorsConfigurationInfo{}
+	}
+
+	allowedOrigins := make([]string, 0)
+	allowedHeaders := make([]string, 0)
+	allowedMethods := make([]string, 0)
+	maxAgeInSeconds := int32(0)
+	allowCredentials := true
+
+	/*for _, attr := range corsConfigRaw {
+		corsConfigAttr := attr.(map[string]interface{})
+
+		allowedOrigins = *utils.ExpandStringSlice(corsConfigAttr["allowed_origins"].([]interface{}))
+		allowedHeaders = *utils.ExpandStringSlice(corsConfigAttr["allowed_headers"].([]interface{}))
+		allowedMethods = *utils.ExpandStringSlice(corsConfigAttr["allowed_methods"].([]interface{}))
+		maxAgeInSeconds = int32(corsConfigAttr["max_age_in_seconds"].(int))
+		allowCredentials = corsConfigAttr["allow_credentials"].(bool)
+	}
+	*/
+
+	corsConfigAttr := corsConfigRaw[0].(map[string]interface{})
+
+	allowedOrigins = *utils.ExpandStringSlice(corsConfigAttr["allowed_origins"].([]interface{}))
+	allowedHeaders = *utils.ExpandStringSlice(corsConfigAttr["allowed_headers"].([]interface{}))
+	allowedMethods = *utils.ExpandStringSlice(corsConfigAttr["allowed_methods"].([]interface{}))
+	maxAgeInSeconds = int32(corsConfigAttr["max_age_in_seconds"].(int))
+	allowCredentials = corsConfigAttr["allow_credentials"].(bool)
+
+	cors := &healthcareapis.ServiceCorsConfigurationInfo{
+		Origins:          &allowedOrigins,
+		Headers:          &allowedHeaders,
+		Methods:          &allowedMethods,
+		MaxAge:           &maxAgeInSeconds,
+		AllowCredentials: &allowCredentials,
+	}
+	return cors
+}
+
+func expandAzureRMhealthcareapisAuthentication(d *schema.ResourceData) *healthcareapis.ServiceAuthenticationConfigurationInfo {
+	authConfigRaw := d.Get("authentication_configuration").([]interface{})
+
+	if len(authConfigRaw) == 0 {
+		return &healthcareapis.ServiceAuthenticationConfigurationInfo{}
+	}
+
+	authority := ""
+	audience := ""
+	smart_proxy_enabled := true
```

Comment:
given we are just setting these below we can remove these?

Reply:
done